diff --git a/libs/glib2/Makefile b/libs/glib2/Makefile
index 5bd0d1cc..966e5567 100644
--- a/libs/glib2/Makefile
+++ b/libs/glib2/Makefile
@@ -18,8 +18,8 @@ PKG_MD5SUM:=5eeb2bfaf78a07be59585e8b6e80b1d6
 
 PKG_BUILD_PARALLEL:=1
 HOST_BUILD_PARALLEL:=1
-PKG_BUILD_DEPENDS:=glib2/host libpthread zlib libintl libffi
-HOST_BUILD_DEPENDS:=gettext-full/host libiconv/host libffi/host
+PKG_BUILD_DEPENDS:=glib2/host libpthread zlib libintl libffi libiconv-full
+HOST_BUILD_DEPENDS:=gettext-full/host libiconv-full/host libffi/host
 PKG_INSTALL:=1
 PKG_USE_MIPS16:=0
 
@@ -46,7 +46,7 @@ endef
 
 HOST_CONFIGURE_ARGS += \
 	--disable-selinux \
-	--with-libiconv=gnu \
+	--with-libiconv=no \
 	--with-pcre=internal
 
 CONFIGURE_ARGS += \
@@ -56,8 +56,8 @@ CONFIGURE_ARGS += \
 	--disable-selinux \
 	--enable-libmount=no \
 	--disable-fam \
-	--with-libiconv=gnu \
-	--with-pcre=internal
+	--with-pcre=internal \
+	--with-libiconv=gnu
 
 CONFIGURE_VARS += \
 	glib_cv_stack_grows=no \
@@ -88,6 +88,7 @@ define Build/InstallDev
 		$(PKG_INSTALL_DIR)/usr/lib/*.{so*,a,la} \
 		$(1)/usr/lib/
 
+	sed -i 's|^Libs:|Libs: -liconv|' $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/*.pc
 	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
 	$(INSTALL_DATA) \
 		$(PKG_INSTALL_DIR)/usr/lib/pkgconfig/*.pc \
