diff --git a/libs/dahdi-tools/Makefile b/libs/dahdi-tools/Makefile
index 572894b..0e137bf 100644
--- a/libs/dahdi-tools/Makefile
+++ b/libs/dahdi-tools/Makefile
@@ -60,5 +60,5 @@ define Build/Compile
 		INSTALL_PREFIX="$(PKG_INSTALL_DIR)" \
 		DAHDI_INCLUDE="$(STAGING_DIR)/usr/include" \
 		CONFIGURE_SILENT="--silent" \
-		dahdi_cfg dahdi_monitor dahdi_scan dahdi_speed dahdi_test fxotune
+		dahdi_cfg dahdi_monitor dahdi_scan dahdi_speed dahdi_test fxotune libtonezone.la
 endef
